completion_aggregator/settings/common.py

```diff
@@ -16,3 +16,14 @@ def plugin_settings(settings):
         'vertical',
     }
     settings.COMPLETION_AGGREGATOR_ASYNC_AGGREGATION = False
+
+    # Names of the batch operations locks
+    settings.COMPLETION_AGGREGATOR_AGGREGATION_LOCK = 'COMPLETION_AGGREGATOR_AGGREGATION_LOCK'
+    settings.COMPLETION_AGGREGATOR_CLEANUP_LOCK = 'COMPLETION_AGGREGATOR_CLEANUP_LOCK'
+
+    # Define how long should the locks be kept. They are released after completing the operation, so there are two
+    # possible scenarios for releasing the lock on timeout:
+    # 1. The management command takes more than 1800s - in this case you should set a higher limit for the lock.
```

Comment:
One alternative would be refreshing the timeout after aggregating each batch using `cache.touch`. This would in turn allow us to reduce the timeout. What do you think?

Reply:
That would be possible, but it would require chaining batch of `update_aggregators` tasks and adding an extra step that releases the lock, because these are asynchronous celery tasks.

Reply:
I meant refreshing the key inside [this loop](https://github.com/open-craft/openedx-completion-aggregator/blob/004337c442a49e44802fa3b48f04dfbe5749cdc4/completion_aggregator/batch.py#L82), but this can be done at a later time if this raises issues.